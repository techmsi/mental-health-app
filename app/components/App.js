
import { Link } from 'react-router';

export default class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
        <nav>
          <Link to="/" activeClassName="active">Home</Link>
          <Link to="/questionnaire">Questionnaire</Link>
          {this.props.children}
        </nav>
    );
  }
}
