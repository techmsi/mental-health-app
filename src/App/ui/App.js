import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { Page } from 'styles/Layout';
import Menu from 'App/ui/Menu';
import AppRoutes from 'App/ui/routes-App';

const App = () => (
  <Page className='app'>
    <BrowserRouter>
      <main>
        <Menu />
        <AppRoutes />
      </main>
    </BrowserRouter>
  </Page>
);

export default App;
