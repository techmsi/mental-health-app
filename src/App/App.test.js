import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from 'App/ui/App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('renders a snapshot of <App />', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
