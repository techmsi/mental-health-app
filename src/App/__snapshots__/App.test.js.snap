// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders a snapshot of <App /> 1`] = `
<section
  className="app sc-bdVaJa eVuSVk"
>
  <main>
    <header
      className="sc-bwzfXH hlWKVP"
    >
      <nav>
        <button
          className="sc-bxivhb cCKQsa"
        >
          <a
            href="/"
            onClick={[Function]}
          >
             Home
          </a>
        </button>
        <button
          className="sc-bxivhb cCKQsa"
        >
          <a
            href="/therapists/list"
            onClick={[Function]}
          >
             Therapists
          </a>
        </button>
      </nav>
    </header>
    <section
      className="introduction"
    >
      <header
        className="sc-bwzfXH hlWKVP"
      >
        <h1>
          Paitent Questionnaire
        </h1>
      </header>
      <p
        className="introduction__text"
      >
        This easy to use patient questionnaire is a self-administered version of the PRIME-MD diagnostic instrument for common mental disorders. The PHQ-9 is the depression module, which scores each of the nine DSM-IV criteria as "0" (not at all) to "3" (nearly every day). It has been validated for use in primary care. It is not a screening tool for depression but it is used to monitor the severity of depression and response to treatment. However, it can be used to make a tentative diagnosis of depression in at-risk populations - eg, those with coronary heart disease or after stroke.
      </p>
      <button
        className="sc-bxivhb cCKQsa"
      >
        <a
          className="introduction__begin"
          href="/questionnaire"
          onClick={[Function]}
        >
          Begin Questionnaire
        </a>
      </button>
    </section>
  </main>
</section>
`;
